gold-phone-input.html

```diff
@@ -181,51 +194,122 @@
       // If there's an initial input, validate it.
       if (this.value)
         this._handleAutoValidate();
+      
+    },
+
+    attached: function() {
+
+      if(this.noCountryCode) {
+        this.set('countryCode', '');
+        Polymer.dom(this).node.querySelector('label').classList.add('label-no-country-code');
+      }
+
+      this.set('maxlength', this.phoneNumberPattern.length);
+
     },
 
     _phoneNumberPatternChanged: function() {
       // Transform the pattern into a regex the iron-input understands.
       var regex = '';
       regex = this.phoneNumberPattern.replace(/\s/g, '\\s');
+      regex = regex.replace(/\(/gi, '\\(');
+      regex = regex.replace(/\)/gi, '\\)');
       regex = regex.replace(/X/gi, '\\d');
       regex = regex.replace(/\+/g, '\\+');
       this.$.input.pattern = regex;
+
+      this.regex = regex;
     },
 
     /**
      * A handler that is called on input
      */
     _onValueChanged: function(value, oldValue) {
+
       // The initial property assignment is handled by `ready`.
-      if (oldValue == undefined || value === oldValue)
+      if (oldValue == undefined || value == undefined) {
+        this.value = '';
         return;
+      }
 
-      //Ensure value is a string
       value = value ? value.toString() : '';
 
       // Keep track of how many dashes the original value has. After
       // reformatting the value, we might gain or lose some of them, which
       // means we have to correctly move the caret to account for the difference.
       var start = this.$.input.selectionStart;
+
       var initialDashesBeforeCaret = value.substr(0, start).split('-').length - 1;
+      var initialSpacesBeforeCaret = value.substr(0, start).split(' ').length - 1;
+      var initialLeftParenBeforeCaret = value.substr(0, start).split('(').length - 1;
+      var initialRightParenBeforeCaret = value.substr(0, start).split(')').length - 1;
+
+      var formattedValue = '';
 
       // Remove any already-applied formatting.
       value = value.replace(/-/g, '');
-      var shouldFormat = value.length <= this.phoneNumberPattern.replace(/-/g, '').length;
-      var formattedValue = '';
+      value = value.replace(/\(/g, '');
```

Comment:
I think all of this would be enormously simplified if you force the format, including the country code to be `xxx-xxxx-xxx`, without the brackets. This isn't that uncommon to be honest, and it would add no code changes